hack/test-integration.sh

```diff
@@ -69,6 +69,7 @@ os::log::start_system_logger
 function exectest() {
 	echo "Running $1..."
 
+	export TEST_ETCD_DIR=$(mktemp ${TMPDIR:-/tmp}/etcd-XXXXXXXX)
```

Comment:
Oh, sorry, didn't see that this was per-test. I'd still like it to be deterministic, so yes, let's name it by the test name